api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -271,7 +271,7 @@ default Transaction newReplaceTableTransaction(
   }
 
   /**
-   * Check whether table exists.
+   * Check whether table or metadata table exists.
```

Comment:
Is checking whether the metadata table exists a new behavior? Since we're changing the comment for the Catalog interface here, it might affect catalogs that have not specifically overridden this function.

Reply:
I believe it's the existing (undocumented) behaviour

Reply:
Thanks! I didn't know about this behavior.

Changing this doc string in the `Catalog` interface will also change the doc string for functions that overrides `tableExists`. For example, the [`EcsCatalog`](https://github.com/apache/iceberg/blob/6501d29b2d46c8d57f46ad646e2daf7d8865f646/dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java#L373-L376) overrides the `tableExists` and inherits the doc string in its javadocs https://iceberg.apache.org/javadoc/1.7.0/org/apache/iceberg/dell/ecs/EcsCatalog.html
but it doesn't look like this behavior is supported in the `EcsCatalog`.

There are a couple more places where this happens https://grep.app/search?q=boolean%20tableExists%28&filter[repo.pattern][0]=apache/iceberg

Reply:
Yea it did cross my mind to be a bit hesitant to change the top level javadoc, I am not sure if metadata tables is a catalog wide concept.  Kevin's suggestion makes sense.

Reply:
I think it's otherwise, all other catalog reuse the default implementation where they tried to load table first as a way to determine whether table exists. The ECSCatalog is the only concrete implementation override this method which might not handle the metadata tables. Partly it was because such behaviour was implicit without unit test or documentation, until @szehon-ho mentioned it in https://github.com/apache/iceberg/pull/11597#discussion_r1853169535

My vote is that we can override the javadoc in ECSCatalog instead, what do you think?

Reply:
I would agree that we don't want to expose the metadata Table concept here.

Reply:
thank you @danielcweeks , revert as suggested. My main concern is that if we dont document this explicitly, the future override of this method in catalog might not even take metadata table into consideration.

Reply:
the metadata table concept only applied to `loadTable` right now 
https://github.com/search?q=repo%3Aapache%2Ficeberg%20isValidMetadataIdentifier&type=code

so its a side effect of [`tableExists` using the underlying `loadTable` function](https://github.com/apache/iceberg/blob/8c04bcb876d46e953ece5b2951c9bd2f361783df/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L279-L286)

on a meta level, its a bit weird to check if a metadata table exists. if the underlying iceberg table exists, I would assume its metadata table also exists. 

that said, I'm in favor of documenting this hidden behavior of `tableExists`

Reply:
Yea I think the main concern with putting on top level javadoc is that , there's a lot of Catalog in the wild that may not (and may not want to) support this.  As @kevinjqliu points out it seems its a bit of an outlier that it happened here (that we should keep for backward compatbiility) and doc can probably be added in HiveCatalog specifically for this PR.